staging/src/k8s.io/apiserver/pkg/endpoints/handlers/apply.go

```diff
@@ -148,3 +149,15 @@ func (p *applyPatcher) applyPatchToCurrentObject(currentObject runtime.Object) (
 
 	return output, nil
 }
+
+func (p *applyPatcher) createNewObject() (runtime.Object, error) {
+	original := p.restPatcher.New()
+	objToCreate, gvk, err := p.codec.Decode(p.patchBytes, &p.kind, original)
+	if err != nil {
+		return nil, transformDecodeError(p.typer, err, original, gvk, p.patchBytes)
+	}
```

Comment:
Should this be defaulted?

Reply:
I think the defaulting is working. The integration test creates a pod which has only containers field populated but many of the other fields are required and get defaulted.